woocommerce/google-listings-and-ads
# Shipping rates UI changes: Move offer free shipping value from useState to Form in Setup MC and Edit Free Listings

Closes #1288 .

Prior to this PR, the "offer free shipping" value is stored in a local `useState` in `FreeShippingCards` component.

In this PR, we move the value out from the local component state into Form state to address issue #1288. `FreeShippingCards` is no longer needed and is removed in this PR.

### Screenshots:

There isn't any major UI change, other than now we allow the offer free shipping to be unchecked (`undefined`):

![image](https://user-images.githubusercontent.com/17420811/156141135-01399b43-852e-42f2-8610-b9af336819bb.png)

The offer free shipping value would be unchecked when:

- Users visit Setup MC for the first time.
- Users edit their existing shipping rates from non-free shipping to all free shipping.

When users add a new non-free shipping, the card would show up with unchecked option.

### Detailed test instructions:

1. Go to Setup MC shipping rate section. When you add a non-free shipping rate, the offer free shipping card should show up with unchecked option.
2. Add a few more free and non-free shipping rates.
3. Remove all the non-free shipping rates, so that you only have free shipping rates. Offer free shipping card should not be visible.
4. Add a non-free shipping rate. The offer free shipping card should show up with unchecked option.
5. Go to Edit Free Listings shipping rate section and repeat the above.

### Additional details:


### Changelog entry
